osu.Game/Screens/SelectV2/BeatmapCarouselFilterGrouping.cs

```diff
@@ -395,6 +395,16 @@ private GroupDefinition defineGroupByCollection(BeatmapInfo beatmap, IEnumerable
             return new GroupDefinition(1, "Not in collection");
         }
 
+        private GroupDefinition defineGroupByOwnMaps(BeatmapInfo beatmap, int? localUserId, string? localUserUsername)
+        {
+            var author = beatmap.BeatmapSet!.Metadata.Author;
+
+            if (author.OnlineID == localUserId || author.Username == localUserUsername)
```

Comment:
this may catch edge cases where a user renamed and another user took their name and ranked a beatmap. but i'm not sure we should care.

Reply:
Applied in https://github.com/ppy/osu/pull/34539/commits/3484d2357b27c859a466fccd273141e999207895